hoomd/mesh.py

```diff
@@ -102,13 +105,33 @@ def triangles(self):
     @triangles.setter
     def triangles(self, triag):
         if self._attached:
-            self._cpp_obj.setTypes(list(self._param_dict['types']))
 
-            self._cpp_obj.setTriangleData(triag)
+            if len(triag) != len(self._type_ids):
+                raise ValueError(
+                    "Number of type_ids do not match number of triangles.")
+            else:
+                self._cpp_obj.setTypes(list(self._param_dict['types']))
```

Comment:
Why not pass the NumPy array?

Reply:
I chose it like this to make it consistent with the rest of the code. For example in `ParticleData.cc`.